### PR TITLE
feat: add a mode type to the pool id

### PR DESCRIPTION
**Pull request by abc3**
This PR introduces the capability to launch multiple pools with different types for the same user. Now, the pool id will be represented as a tuple consisting of the tenant, username and mode type (session or transaction). 

This PR also includes bug fixes and refactoring

### CODE REVIEW

File: lib/supavisor/client_handler.ex [hunk 1]
```diff
@@ -6,18 +6,15 @@ defmodule Supavisor.ClientHandler do
...
   require Logger
 
   @behaviour :ranch_protocol
   @behaviour :gen_statem
 
   alias Supavisor.DbHandler, as: Db
-  alias Supavisor.Helpers, as: H
-  alias Supavisor.{Tenants, Protocol.Server, Monitoring.Telem}
+  alias Supavisor, as: S
+  alias S.Helpers, as: H
```
Comment (abc3) [inline lib/supavisor/client_handler.ex:16]:
```suggestion
  alias Supavisor.Helpers, as: H
  alias Supavisor, as: S

```
```suggestion
  alias Supavisor, as: S
  alias S.Helpers, as: H
```
```suggestion
  alias Supavisor.Helpers, as: H
  alias Supavisor, as: S
```

File: lib/supavisor/manager.ex [hunk 2]
```diff
@@ -3,11 +3,10 @@ defmodule Supavisor.Manager do
   use GenServer, restart: :transient
   require Logger
 
-  alias Supavisor.PromEx.Plugins.Tenant, as: M
   alias Supavisor.Protocol.Server
   alias Supavisor.Tenants
 
-  @check_timeout 120_000
+  @check_timeout 10_000
```
Comment (abc3) [inline lib/supavisor/manager.ex:9]:
```suggestion
  @check_timeout 120_000
```

File: lib/supavisor.ex [hunk 3]
```diff
@@ -104,37 +111,63 @@ defmodule Supavisor do
...
+    [node() | Node.list()]
+    |> Task.async_stream(
+      fn node ->
+        %{node => :rpc.call(node, Supavisor, :dirty_terminate, [tenant], 60_000)}
+      end,
+      timeout: :infinity
+    )
+    |> Enum.reduce([], fn resp, acc ->
+      [inspect(resp) | acc]
+    end)
```
Comment (josevalim) [inline lib/supavisor.ex:124]: This is completely fine, but you may also consider using https://www.erlang.org/doc/man/erpc#multicall-4 - which will separate the separate calls without spawning new tasks. Also note that `Task.async_stream` has a concurrency limit, which I don't think you care here?

File: Makefile [hunk 4]
```diff
@@ -24,8 +24,10 @@ dev.node2:
 	REGION=eu \
 	SECRET_KEY_BASE="dev" \
 	CLUSTER_POSTGRES="true" \
+	PROXY_PORT_SESSION="5442" \
+	PROXY_PORT_TRANSACTION="6553" \
```
Comment (josevalim) [inline Makefile:28]: Notice this env var is already set earlier.

File: config/test.exs [hunk 5]
```diff
@@ -6,6 +6,7 @@ config :supavisor,
   api_jwt_secret: "dev",
   metrics_jwt_secret: "dev",
   jwt_claim_validators: %{},
+  proxy_port_session: System.get_env("PROXY_PORT_TRANSACTION", "7653") |> String.to_integer(),
```
Comment (josevalim) [inline config/test.exs:9]: Apologies, I may be missing something obvious, but is this config used anywhere? Also, the env var does not match the config key, I am not sure if it is intentional.

**Review by josevalim — COMMENTED** [review]
I have added some minor nits, but it looks great to me!

My understanding is that we already had two ports before, but they used the same pool. Now they are different pool. Some other important characteristics:

1. Pools are started lazily. This means if you run both session and transaction, but only ever use one, then only one is ever created (please confirm if my understanding is correct)

2. However, connections are started eagerly, so if you are using both, then you get the double of connections, but that's most likely what you want anyway

I can see people wanting different sizes for each pool, but that can be a conversation for another day. :)

File: lib/supavisor.ex (hunk 3 above)
Comment (abc3) [reply lib/supavisor.ex:124]: good call 👍

File: Makefile (hunk 4 above)
Comment (abc3) [reply Makefile:28]: oops

File: config/test.exs (hunk 5 above)
Comment (abc3) [reply config/test.exs:9]: 🤦

**Comment by abc3** [comment]
@josevalim sorry for the late response

> Pools are started lazily. This means if you run both session and transaction, but only ever use one, then only one is ever created (please confirm if my understanding is correct)

Yes, correct

> However, connections are started eagerly, so if you are using both, then you get the double of connections, but that's most likely what you want anyway

Yes, exactly!

> I can see people wanting different sizes for each pool, but that can be a conversation for another day. :)

Hmm, that is a good suggestion! I will bring that idea to the team 👍

File: lib/supavisor.ex (hunk 3 above)
Comment (abc3) [reply lib/supavisor.ex]: updated https://github.com/supabase/supavisor/pull/160/commits/a6614b363ba4c481d329cd54d32a70cca0596072

**Review by chasers — APPROVED** [review]

**Pull request merged**